### PR TITLE
User/Register Code

### PR DESCRIPTION
**Pull request by ravensears**

### CODE REVIEW

File: lib/user.rb [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+class User
+  def self.create(name:, username:, email:, password:)
+    result = DatabaseConnection.query(
+      "INSERT INTO users (name, username, email, password) VALUES($1, $2, $3, $4) RETURNING id, name;", 
+      [name, username, email, password]
+    )
+    User.new(id: result[0]['id'], name: result[0]['name'])
+  end
+  
+  attr_reader :id, :name
```
Comment (ravensears) [inline lib/user.rb:12]: I would put the attr_reader and initialise at the top of the class for visibility x

File: spec/features/register_spec.rb [hunk 2]
```diff
@@ -0,0 +1,16 @@
+# As a user,
```
Comment (ravensears) [inline spec/features/register_spec.rb:1]: I would remove the user story x

File: app.rb [hunk 3]
```diff
@@ -1,14 +1,27 @@
...
   end
 
   get '/' do
-    'Hello World'
+    @user = session[:user]
+    erb :index
+  end
+
+  get '/users/new' do
+    erb :"users/new"
```
Comment (fpmrqs) [inline app.rb:18]: Nice use of folders for paths, we probably should do the same with spaces!

**Review by ravensears — COMMENTED** [review]
Looks great, well done y'all x

File: spec/user_spec.rb [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+require 'database_helper'
+
+describe '.create' do
+  it 'creates a new user' do
+    user = User.create(name: 'test', username: 'test1', email: 'test@example.com', password: 'password123')
+
+    # persisted_data = persisted_data(table: :users, id: user.id)
+
+    expect(user).to be_a User
+    expect(user.name).to eq 'test'
```
Comment (fpmrqs) [inline spec/user_spec.rb:11]: We might need to create more 'expect's for each user property (id, email etc), apart from that it's all good

**Pull request merged**